src/main/java/de/tum/in/www1/artemis/service/DataExportService.java

```diff
@@ -0,0 +1,567 @@
+        for (var participation : exercise.getStudentParticipations()) {
+            for (var submission : participation.getSubmissions()) {
+                createSubmissionCsvFile(submission, exerciseDir);
+                if (submission instanceof FileUploadSubmission) {
+                    copyFileUploadSubmissionFile(FileUploadSubmission.buildFilePath(exercise.getId(), submission.getId()), exerciseDir);
+                }
+                else if (submission instanceof TextSubmission textSubmission) {
+                    storeTextSubmissionContent(textSubmission, exerciseDir);
+                }
+                else if (submission instanceof ModelingSubmission modelingSubmission) {
+                    storeModelingSubmissionContent(modelingSubmission, exerciseDir);
+                }
+                else if (submission instanceof QuizSubmission) {
+                    createQuizAnswersExport((QuizExercise) exercise, participation, exerciseDir);
+                }
+
+                createResultsTxtFile(submission, exerciseDir);
+            }
+        }
+    }
+
+    private void createQuizAnswersExport(QuizExercise quizExercise, StudentParticipation participation, Path outputDir) throws IOException {
+        Set<QuizQuestion> quizQuestions = quizQuestionRepository.getQuizQuestionsByExerciseId(quizExercise.getId());
+        QuizSubmission quizSubmission;
+
+        for (var submission : participation.getSubmissions()) {
+            quizSubmission = quizSubmissionRepository.findWithEagerSubmittedAnswersById(submission.getId());
+            List<String> multipleChoiceQuestionsSubmissions = new ArrayList<>();
+            List<String> shortAnswerQuestionsSubmissions = new ArrayList<>();
+            for (var question : quizQuestions) {
+                var submittedAnswer = quizSubmission.getSubmittedAnswerForQuestion(question);
+                // if this question wasn't answered, the submitted answer is null
+                if (submittedAnswer != null) {
+                    if (submittedAnswer instanceof DragAndDropSubmittedAnswer dragAndDropSubmittedAnswer) {
+                        dragAndDropQuizAnswerConversionService.convertDragAndDropQuizAnswerAndStoreAsPdf(dragAndDropSubmittedAnswer, outputDir);
+                    }
+                    else if (submittedAnswer instanceof ShortAnswerSubmittedAnswer shortAnswerSubmittedAnswer) {
+                        shortAnswerQuestionsSubmissions.add(createExportForShortAnswerQuestion(shortAnswerSubmittedAnswer));
+                    }
+                    else if (submittedAnswer instanceof MultipleChoiceSubmittedAnswer multipleChoiceSubmittedAnswer) {
+                        multipleChoiceQuestionsSubmissions.add(createExportForMultipleChoiceAnswerQuestion(multipleChoiceSubmittedAnswer));
+                    }
+                }
+            }
+            if (!multipleChoiceQuestionsSubmissions.isEmpty()) {
```

Comment:
Since you only check for emptiness, wouldn't a boolean be enough instead of storing a whole list?